deeppavlov/core/commands/train.py

```diff
@@ -83,21 +83,43 @@ def fit_chainer(config: dict, iterator: Union[DataLearningIterator, DataFittingI
             if len(component_config['fit_on']) == 1:
                 preprocessed = [preprocessed]
 
-            component.fit(*preprocessed)
+            result = component.fit(*preprocessed)
+            if result is not None and config['train'].get('tensorboard_log_dir') is not None:
+                import tensorflow as tf
+                tb_log_dir = expand_path(config['train']['tensorboard_log_dir'])
+                writer = tf.summary.FileWriter(str(tb_log_dir / 'fit_log'))
+
+                for name, scores in result.items():
+                    for i, score in enumerate(scores):
+                        metric_sum = tf.Summary(value=[tf.Summary.Value(tag='fit/' + name,
+                                                                        simple_value=score), ])
+                        writer.add_summary(metric_sum, i)
+
             component.save()
 
-        if 'fit_on_batch' in component_config:
+        if 'fit_on_batches' in component_config:
```

Comment:
it would be great if both options would work, but for an older one a deprecation warning would be thrown